{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: Streamlit in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (0.84.0)\n",
      "Requirement already satisfied: attrs in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (20.3.0)\n",
      "Requirement already satisfied: click<8.0,>=7.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (7.1.2)\n",
      "Requirement already satisfied: pandas>=0.21.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (1.1.3)\n",
      "Requirement already satisfied: toml in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (0.10.1)\n",
      "Requirement already satisfied: protobuf!=3.11,>=3.6.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (3.14.0)\n",
      "Requirement already satisfied: requests in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (2.24.0)\n",
      "Requirement already satisfied: pillow>=6.2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (8.0.1)\n",
      "Requirement already satisfied: blinker in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (1.4)\n",
      "Requirement already satisfied: python-dateutil in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (2.8.1)\n",
      "Requirement already satisfied: pydeck>=0.1.dev5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (0.6.2)\n",
      "Requirement already satisfied: pyarrow; python_version < \"3.9\" in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (4.0.1)\n",
      "Requirement already satisfied: cachetools>=4.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (4.1.1)\n",
      "Requirement already satisfied: base58 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (2.1.0)\n",
      "Requirement already satisfied: tzlocal in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (2.1)\n",
      "Requirement already satisfied: validators in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (0.18.2)\n",
      "Requirement already satisfied: astor in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (0.8.1)\n",
      "Requirement already satisfied: packaging in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (20.4)\n",
      "Requirement already satisfied: numpy in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (1.18.5)\n",
      "Requirement already satisfied: altair>=3.2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (4.1.0)\n",
      "Requirement already satisfied: tornado>=5.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (6.0.4)\n",
      "Requirement already satisfied: gitpython in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Streamlit) (3.1.18)\n",
      "Requirement already satisfied: pytz>=2017.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pandas>=0.21.0->Streamlit) (2020.1)\n",
      "Requirement already satisfied: six>=1.9 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from protobuf!=3.11,>=3.6.0->Streamlit) (1.15.0)\n",
      "Requirement already satisfied: idna<3,>=2.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from requests->Streamlit) (2.10)\n",
      "Requirement already satisfied: chardet<4,>=3.0.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from requests->Streamlit) (3.0.4)\n",
      "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from requests->Streamlit) (1.25.11)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from requests->Streamlit) (2020.6.20)\n",
      "Requirement already satisfied: jinja2>=2.10.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pydeck>=0.1.dev5->Streamlit) (2.11.2)\n",
      "Requirement already satisfied: ipykernel>=5.1.2; python_version >= \"3.4\" in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pydeck>=0.1.dev5->Streamlit) (5.3.4)\n",
      "Requirement already satisfied: traitlets>=4.3.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pydeck>=0.1.dev5->Streamlit) (5.0.5)\n",
      "Requirement already satisfied: ipywidgets>=7.0.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pydeck>=0.1.dev5->Streamlit) (7.5.1)\n",
      "Requirement already satisfied: decorator>=3.4.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from validators->Streamlit) (4.4.2)\n",
      "Requirement already satisfied: pyparsing>=2.0.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from packaging->Streamlit) (2.4.7)\n",
      "Requirement already satisfied: toolz in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from altair>=3.2.0->Streamlit) (0.11.1)\n",
      "Requirement already satisfied: jsonschema in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from altair>=3.2.0->Streamlit) (3.2.0)\n",
      "Requirement already satisfied: entrypoints in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from altair>=3.2.0->Streamlit) (0.3)\n",
      "Requirement already satisfied: gitdb<5,>=4.0.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from gitpython->Streamlit) (4.0.7)\n",
      "Requirement already satisfied: MarkupSafe>=0.23 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from jinja2>=2.10.1->pydeck>=0.1.dev5->Streamlit) (1.1.1)\n",
      "Requirement already satisfied: jupyter-client in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from ipykernel>=5.1.2; python_version >= \"3.4\"->pydeck>=0.1.dev5->Streamlit) (6.1.7)\n",
      "Requirement already satisfied: appnope; platform_system == \"Darwin\" in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from ipykernel>=5.1.2; python_version >= \"3.4\"->pydeck>=0.1.dev5->Streamlit) (0.1.0)\n",
      "Requirement already satisfied: ipython>=5.0.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from ipykernel>=5.1.2; python_version >= \"3.4\"->pydeck>=0.1.dev5->Streamlit) (7.19.0)\n",
      "Requirement already satisfied: ipython-genutils in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from traitlets>=4.3.2->pydeck>=0.1.dev5->Streamlit) (0.2.0)\n",
      "Requirement already satisfied: widgetsnbextension~=3.5.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (3.5.1)\n",
      "Requirement already satisfied: nbformat>=4.2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (5.0.8)\n",
      "Requirement already satisfied: pyrsistent>=0.14.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from jsonschema->altair>=3.2.0->Streamlit) (0.17.3)\n",
      "Requirement already satisfied: setuptools in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from jsonschema->altair>=3.2.0->Streamlit) (50.3.1.post20201107)\n",
      "Requirement already satisfied: smmap<5,>=3.0.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from gitdb<5,>=4.0.1->gitpython->Streamlit) (4.0.0)\n",
      "Requirement already satisfied: jupyter-core>=4.6.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from jupyter-client->ipykernel>=5.1.2; python_version >= \"3.4\"->pydeck>=0.1.dev5->Streamlit) (4.6.3)\n",
      "Requirement already satisfied: pyzmq>=13 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from jupyter-client->ipykernel>=5.1.2; python_version >= \"3.4\"->pydeck>=0.1.dev5->Streamlit) (19.0.2)\n",
      "Requirement already satisfied: jedi>=0.10 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from ipython>=5.0.0->ipykernel>=5.1.2; python_version >= \"3.4\"->pydeck>=0.1.dev5->Streamlit) (0.17.1)\n",
      "Requirement already satisfied: pygments in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from ipython>=5.0.0->ipykernel>=5.1.2; python_version >= \"3.4\"->pydeck>=0.1.dev5->Streamlit) (2.7.2)\n",
      "Requirement already satisfied: backcall in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from ipython>=5.0.0->ipykernel>=5.1.2; python_version >= \"3.4\"->pydeck>=0.1.dev5->Streamlit) (0.2.0)\n",
      "Requirement already satisfied: pickleshare in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from ipython>=5.0.0->ipykernel>=5.1.2; python_version >= \"3.4\"->pydeck>=0.1.dev5->Streamlit) (0.7.5)\n",
      "Requirement already satisfied: pexpect>4.3; sys_platform != \"win32\" in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from ipython>=5.0.0->ipykernel>=5.1.2; python_version >= \"3.4\"->pydeck>=0.1.dev5->Streamlit) (4.8.0)\n",
      "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from ipython>=5.0.0->ipykernel>=5.1.2; python_version >= \"3.4\"->pydeck>=0.1.dev5->Streamlit) (3.0.8)\n",
      "Requirement already satisfied: notebook>=4.4.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (6.1.4)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: parso<0.8.0,>=0.7.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from jedi>=0.10->ipython>=5.0.0->ipykernel>=5.1.2; python_version >= \"3.4\"->pydeck>=0.1.dev5->Streamlit) (0.7.0)\n",
      "Requirement already satisfied: ptyprocess>=0.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pexpect>4.3; sys_platform != \"win32\"->ipython>=5.0.0->ipykernel>=5.1.2; python_version >= \"3.4\"->pydeck>=0.1.dev5->Streamlit) (0.6.0)\n",
      "Requirement already satisfied: wcwidth in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=5.0.0->ipykernel>=5.1.2; python_version >= \"3.4\"->pydeck>=0.1.dev5->Streamlit) (0.2.5)\n",
      "Requirement already satisfied: Send2Trash in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (1.5.0)\n",
      "Requirement already satisfied: terminado>=0.8.3 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (0.9.1)\n",
      "Requirement already satisfied: prometheus-client in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (0.8.0)\n",
      "Requirement already satisfied: nbconvert in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (6.0.7)\n",
      "Requirement already satisfied: argon2-cffi in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (20.1.0)\n",
      "Requirement already satisfied: defusedxml in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (0.6.0)\n",
      "Requirement already satisfied: bleach in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (3.2.1)\n",
      "Requirement already satisfied: jupyterlab-pygments in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (0.1.2)\n",
      "Requirement already satisfied: nbclient<0.6.0,>=0.5.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (0.5.1)\n",
      "Requirement already satisfied: testpath in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (0.4.4)\n",
      "Requirement already satisfied: mistune<2,>=0.8.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (0.8.4)\n",
      "Requirement already satisfied: pandocfilters>=1.4.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (1.4.3)\n",
      "Requirement already satisfied: cffi>=1.0.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (1.14.3)\n",
      "Requirement already satisfied: webencodings in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from bleach->nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (0.5.1)\n",
      "Requirement already satisfied: async-generator in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from nbclient<0.6.0,>=0.5.0->nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (1.10)\n",
      "Requirement already satisfied: nest-asyncio in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from nbclient<0.6.0,>=0.5.0->nbconvert->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (1.4.2)\n",
      "Requirement already satisfied: pycparser in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from cffi>=1.0.0->argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.5.0->ipywidgets>=7.0.0->pydeck>=0.1.dev5->Streamlit) (2.20)\n"
     ]
    }
   ],
   "source": [
    "!pip install Streamlit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: plotly in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (4.14.1)\r\n",
      "Requirement already satisfied: retrying>=1.3.3 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from plotly) (1.3.3)\r\n",
      "Requirement already satisfied: six in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from plotly) (1.15.0)\r\n"
     ]
    }
   ],
   "source": [
    "!pip install plotly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: chart-studio in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (1.1.0)\n",
      "Requirement already satisfied: retrying>=1.3.3 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from chart-studio) (1.3.3)\n",
      "Requirement already satisfied: plotly in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from chart-studio) (4.14.1)\n",
      "Requirement already satisfied: six in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from chart-studio) (1.15.0)\n",
      "Requirement already satisfied: requests in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from chart-studio) (2.24.0)\n",
      "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from requests->chart-studio) (1.25.11)\n",
      "Requirement already satisfied: chardet<4,>=3.0.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from requests->chart-studio) (3.0.4)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from requests->chart-studio) (2020.6.20)\n",
      "Requirement already satisfied: idna<3,>=2.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from requests->chart-studio) (2.10)\n"
     ]
    }
   ],
   "source": [
    "!pip install chart-studio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting Fbprophet\n",
      "  Using cached fbprophet-0.7.1.tar.gz (64 kB)\n",
      "Requirement already satisfied: Cython>=0.22 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Fbprophet) (0.29.21)\n",
      "Requirement already satisfied: cmdstanpy==0.9.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Fbprophet) (0.9.5)\n",
      "Requirement already satisfied: pystan>=2.14 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Fbprophet) (3.2.0)\n",
      "Requirement already satisfied: numpy>=1.15.4 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Fbprophet) (1.18.5)\n",
      "Requirement already satisfied: pandas>=1.0.4 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Fbprophet) (1.1.3)\n",
      "Requirement already satisfied: matplotlib>=2.0.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Fbprophet) (3.3.2)\n",
      "Requirement already satisfied: LunarCalendar>=0.0.9 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Fbprophet) (0.0.9)\n",
      "Requirement already satisfied: convertdate>=2.1.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Fbprophet) (2.3.2)\n",
      "Requirement already satisfied: holidays>=0.10.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Fbprophet) (0.11.1)\n",
      "Requirement already satisfied: setuptools-git>=1.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Fbprophet) (1.2)\n",
      "Requirement already satisfied: python-dateutil>=2.8.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Fbprophet) (2.8.1)\n",
      "Requirement already satisfied: tqdm>=4.36.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from Fbprophet) (4.50.2)\n",
      "Requirement already satisfied: pysimdjson<4.0,>=3.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan>=2.14->Fbprophet) (3.2.0)\n",
      "Requirement already satisfied: aiohttp<4.0,>=3.6 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan>=2.14->Fbprophet) (3.7.4.post0)\n",
      "Requirement already satisfied: httpstan<4.6,>=4.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan>=2.14->Fbprophet) (4.5.0)\n",
      "Requirement already satisfied: clikit<0.7,>=0.6 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan>=2.14->Fbprophet) (0.6.2)\n",
      "Requirement already satisfied: pytz>=2017.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pandas>=1.0.4->Fbprophet) (2020.1)\n",
      "Requirement already satisfied: certifi>=2020.06.20 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0.0->Fbprophet) (2020.6.20)\n",
      "Requirement already satisfied: pillow>=6.2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0.0->Fbprophet) (8.0.1)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0.0->Fbprophet) (1.3.0)\n",
      "Requirement already satisfied: cycler>=0.10 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0.0->Fbprophet) (0.10.0)\n",
      "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0.0->Fbprophet) (2.4.7)\n",
      "Requirement already satisfied: ephem>=3.7.5.3 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from LunarCalendar>=0.0.9->Fbprophet) (4.0.0.2)\n",
      "Requirement already satisfied: pymeeus<=1,>=0.3.13 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from convertdate>=2.1.2->Fbprophet) (0.5.11)\n",
      "Requirement already satisfied: six in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from holidays>=0.10.2->Fbprophet) (1.15.0)\n",
      "Requirement already satisfied: hijri-converter in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from holidays>=0.10.2->Fbprophet) (2.1.3)\n",
      "Requirement already satisfied: korean-lunar-calendar in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from holidays>=0.10.2->Fbprophet) (0.2.1)\n",
      "Requirement already satisfied: multidict<7.0,>=4.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan>=2.14->Fbprophet) (5.1.0)\n",
      "Requirement already satisfied: async-timeout<4.0,>=3.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan>=2.14->Fbprophet) (3.0.1)\n",
      "Requirement already satisfied: typing-extensions>=3.6.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan>=2.14->Fbprophet) (3.7.4.3)\n",
      "Requirement already satisfied: chardet<5.0,>=2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan>=2.14->Fbprophet) (3.0.4)\n",
      "Requirement already satisfied: yarl<2.0,>=1.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan>=2.14->Fbprophet) (1.6.3)\n",
      "Requirement already satisfied: attrs>=17.3.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan>=2.14->Fbprophet) (20.3.0)\n",
      "Requirement already satisfied: lz4<4.0,>=3.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan>=2.14->Fbprophet) (3.1.3)\n",
      "Requirement already satisfied: appdirs<2.0,>=1.4 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan>=2.14->Fbprophet) (1.4.4)\n",
      "Requirement already satisfied: webargs<8.0,>=7.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan>=2.14->Fbprophet) (7.0.1)\n",
      "Requirement already satisfied: marshmallow<4.0,>=3.10 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan>=2.14->Fbprophet) (3.12.1)\n",
      "Requirement already satisfied: setuptools>=41.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan>=2.14->Fbprophet) (50.3.1.post20201107)\n",
      "Requirement already satisfied: crashtest<0.4.0,>=0.3.0; python_version >= \"3.6\" and python_version < \"4.0\" in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from clikit<0.7,>=0.6->pystan>=2.14->Fbprophet) (0.3.1)\n",
      "Requirement already satisfied: pylev<2.0,>=1.3 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from clikit<0.7,>=0.6->pystan>=2.14->Fbprophet) (1.4.0)\n",
      "Requirement already satisfied: pastel<0.3.0,>=0.2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from clikit<0.7,>=0.6->pystan>=2.14->Fbprophet) (0.2.1)\n",
      "Requirement already satisfied: idna>=2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from yarl<2.0,>=1.0->aiohttp<4.0,>=3.6->pystan>=2.14->Fbprophet) (2.10)\n",
      "Building wheels for collected packages: Fbprophet\n",
      "  Building wheel for Fbprophet (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /Users/l.sannecompethance.fr/opt/anaconda3/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/setup.py'\"'\"'; __file__='\"'\"'/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' bdist_wheel -d /private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-wheel-xt2qajoi\n",
      "       cwd: /private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/\n",
      "  Complete output (38 lines):\n",
      "  running bdist_wheel\n",
      "  running build\n",
      "  running build_py\n",
      "  creating build\n",
      "  creating build/lib\n",
      "  creating build/lib/fbprophet\n",
      "  creating build/lib/fbprophet/stan_model\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/setup.py\", line 122, in <module>\n",
      "      setup(\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages/setuptools/__init__.py\", line 153, in setup\n",
      "      return distutils.core.setup(**attrs)\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/core.py\", line 148, in setup\n",
      "      dist.run_commands()\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 966, in run_commands\n",
      "      self.run_command(cmd)\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 985, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages/wheel/bdist_wheel.py\", line 299, in run\n",
      "      self.run_command('build')\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/cmd.py\", line 313, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 985, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/command/build.py\", line 135, in run\n",
      "      self.run_command(cmd_name)\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/cmd.py\", line 313, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 985, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/setup.py\", line 48, in run\n",
      "      build_models(target_dir)\n",
      "    File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/setup.py\", line 38, in build_models\n",
      "      StanBackendEnum.get_backend_class(backend).build_model(target_dir, MODEL_DIR)\n",
      "    File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/fbprophet/models.py\", line 209, in build_model\n",
      "      import pystan\n",
      "  ModuleNotFoundError: No module named 'pystan'\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[31m  ERROR: Failed building wheel for Fbprophet\u001b[0m\n",
      "\u001b[?25h  Running setup.py clean for Fbprophet\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed to build Fbprophet\n",
      "Installing collected packages: Fbprophet\n",
      "    Running setup.py install for Fbprophet ... \u001b[?25lerror\n",
      "\u001b[31m    ERROR: Command errored out with exit status 1:\n",
      "     command: /Users/l.sannecompethance.fr/opt/anaconda3/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/setup.py'\"'\"'; __file__='\"'\"'/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record /private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-record-lheadz25/install-record.txt --single-version-externally-managed --compile --install-headers /Users/l.sannecompethance.fr/opt/anaconda3/include/python3.8/Fbprophet\n",
      "         cwd: /private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/\n",
      "    Complete output (40 lines):\n",
      "    running install\n",
      "    running build\n",
      "    running build_py\n",
      "    creating build\n",
      "    creating build/lib\n",
      "    creating build/lib/fbprophet\n",
      "    creating build/lib/fbprophet/stan_model\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/setup.py\", line 122, in <module>\n",
      "        setup(\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages/setuptools/__init__.py\", line 153, in setup\n",
      "        return distutils.core.setup(**attrs)\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/core.py\", line 148, in setup\n",
      "        dist.run_commands()\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 966, in run_commands\n",
      "        self.run_command(cmd)\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 985, in run_command\n",
      "        cmd_obj.run()\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages/setuptools/command/install.py\", line 61, in run\n",
      "        return orig.install.run(self)\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/command/install.py\", line 545, in run\n",
      "        self.run_command('build')\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/cmd.py\", line 313, in run_command\n",
      "        self.distribution.run_command(command)\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 985, in run_command\n",
      "        cmd_obj.run()\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/command/build.py\", line 135, in run\n",
      "        self.run_command(cmd_name)\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/cmd.py\", line 313, in run_command\n",
      "        self.distribution.run_command(command)\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 985, in run_command\n",
      "        cmd_obj.run()\n",
      "      File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/setup.py\", line 48, in run\n",
      "        build_models(target_dir)\n",
      "      File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/setup.py\", line 38, in build_models\n",
      "        StanBackendEnum.get_backend_class(backend).build_model(target_dir, MODEL_DIR)\n",
      "      File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/fbprophet/models.py\", line 209, in build_model\n",
      "        import pystan\n",
      "    ModuleNotFoundError: No module named 'pystan'\n",
      "    ----------------------------------------\u001b[0m\n",
      "\u001b[31mERROR: Command errored out with exit status 1: /Users/l.sannecompethance.fr/opt/anaconda3/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/setup.py'\"'\"'; __file__='\"'\"'/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-bymi0y7i/fbprophet/setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record /private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-record-lheadz25/install-record.txt --single-version-externally-managed --compile --install-headers /Users/l.sannecompethance.fr/opt/anaconda3/include/python3.8/Fbprophet Check the logs for full command output.\u001b[0m\n",
      "\u001b[?25h"
     ]
    }
   ],
   "source": [
    "!pip install Fbprophet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pystan in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (3.2.0)\n",
      "Requirement already satisfied: httpstan<4.6,>=4.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan) (4.5.0)\n",
      "Requirement already satisfied: aiohttp<4.0,>=3.6 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan) (3.7.4.post0)\n",
      "Requirement already satisfied: clikit<0.7,>=0.6 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan) (0.6.2)\n",
      "Requirement already satisfied: numpy<2.0,>=1.7 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan) (1.18.5)\n",
      "Requirement already satisfied: pysimdjson<4.0,>=3.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan) (3.2.0)\n",
      "Requirement already satisfied: webargs<8.0,>=7.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan) (7.0.1)\n",
      "Requirement already satisfied: lz4<4.0,>=3.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan) (3.1.3)\n",
      "Requirement already satisfied: appdirs<2.0,>=1.4 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan) (1.4.4)\n",
      "Requirement already satisfied: setuptools>=41.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan) (50.3.1.post20201107)\n",
      "Requirement already satisfied: marshmallow<4.0,>=3.10 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan) (3.12.1)\n",
      "Requirement already satisfied: async-timeout<4.0,>=3.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan) (3.0.1)\n",
      "Requirement already satisfied: chardet<5.0,>=2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan) (3.0.4)\n",
      "Requirement already satisfied: attrs>=17.3.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan) (20.3.0)\n",
      "Requirement already satisfied: multidict<7.0,>=4.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan) (5.1.0)\n",
      "Requirement already satisfied: typing-extensions>=3.6.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan) (3.7.4.3)\n",
      "Requirement already satisfied: yarl<2.0,>=1.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan) (1.6.3)\n",
      "Requirement already satisfied: pastel<0.3.0,>=0.2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from clikit<0.7,>=0.6->pystan) (0.2.1)\n",
      "Requirement already satisfied: crashtest<0.4.0,>=0.3.0; python_version >= \"3.6\" and python_version < \"4.0\" in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from clikit<0.7,>=0.6->pystan) (0.3.1)\n",
      "Requirement already satisfied: pylev<2.0,>=1.3 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from clikit<0.7,>=0.6->pystan) (1.4.0)\n",
      "Requirement already satisfied: idna>=2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from yarl<2.0,>=1.0->aiohttp<4.0,>=3.6->pystan) (2.10)\n"
     ]
    }
   ],
   "source": [
    "!pip install pystan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already up-to-date: pystan in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (3.2.0)\n",
      "Requirement already satisfied, skipping upgrade: numpy<2.0,>=1.7 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan) (1.18.5)\n",
      "Requirement already satisfied, skipping upgrade: clikit<0.7,>=0.6 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan) (0.6.2)\n",
      "Requirement already satisfied, skipping upgrade: pysimdjson<4.0,>=3.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan) (3.2.0)\n",
      "Requirement already satisfied, skipping upgrade: aiohttp<4.0,>=3.6 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan) (3.7.4.post0)\n",
      "Requirement already satisfied, skipping upgrade: httpstan<4.6,>=4.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan) (4.5.0)\n",
      "Requirement already satisfied, skipping upgrade: pastel<0.3.0,>=0.2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from clikit<0.7,>=0.6->pystan) (0.2.1)\n",
      "Requirement already satisfied, skipping upgrade: pylev<2.0,>=1.3 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from clikit<0.7,>=0.6->pystan) (1.4.0)\n",
      "Requirement already satisfied, skipping upgrade: crashtest<0.4.0,>=0.3.0; python_version >= \"3.6\" and python_version < \"4.0\" in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from clikit<0.7,>=0.6->pystan) (0.3.1)\n",
      "Requirement already satisfied, skipping upgrade: attrs>=17.3.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan) (20.3.0)\n",
      "Requirement already satisfied, skipping upgrade: yarl<2.0,>=1.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan) (1.6.3)\n",
      "Requirement already satisfied, skipping upgrade: async-timeout<4.0,>=3.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan) (3.0.1)\n",
      "Requirement already satisfied, skipping upgrade: multidict<7.0,>=4.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan) (5.1.0)\n",
      "Requirement already satisfied, skipping upgrade: typing-extensions>=3.6.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan) (3.7.4.3)\n",
      "Requirement already satisfied, skipping upgrade: chardet<5.0,>=2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan) (3.0.4)\n",
      "Requirement already satisfied, skipping upgrade: marshmallow<4.0,>=3.10 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan) (3.12.1)\n",
      "Requirement already satisfied, skipping upgrade: appdirs<2.0,>=1.4 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan) (1.4.4)\n",
      "Requirement already satisfied, skipping upgrade: setuptools>=41.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan) (50.3.1.post20201107)\n",
      "Requirement already satisfied, skipping upgrade: lz4<4.0,>=3.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan) (3.1.3)\n",
      "Requirement already satisfied, skipping upgrade: webargs<8.0,>=7.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan) (7.0.1)\n",
      "Requirement already satisfied, skipping upgrade: idna>=2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from yarl<2.0,>=1.0->aiohttp<4.0,>=3.6->pystan) (2.10)\n"
     ]
    }
   ],
   "source": [
    "!pip install --upgrade pystan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting fbprophet\n",
      "  Using cached fbprophet-0.7.1.tar.gz (64 kB)\n",
      "Requirement already satisfied: Cython>=0.22 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from fbprophet) (0.29.21)\n",
      "Requirement already satisfied: cmdstanpy==0.9.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from fbprophet) (0.9.5)\n",
      "Requirement already satisfied: pystan>=2.14 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from fbprophet) (3.2.0)\n",
      "Requirement already satisfied: numpy>=1.15.4 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from fbprophet) (1.18.5)\n",
      "Requirement already satisfied: pandas>=1.0.4 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from fbprophet) (1.1.3)\n",
      "Requirement already satisfied: matplotlib>=2.0.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from fbprophet) (3.3.2)\n",
      "Requirement already satisfied: LunarCalendar>=0.0.9 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from fbprophet) (0.0.9)\n",
      "Requirement already satisfied: convertdate>=2.1.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from fbprophet) (2.3.2)\n",
      "Requirement already satisfied: holidays>=0.10.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from fbprophet) (0.11.1)\n",
      "Requirement already satisfied: setuptools-git>=1.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from fbprophet) (1.2)\n",
      "Requirement already satisfied: python-dateutil>=2.8.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from fbprophet) (2.8.1)\n",
      "Requirement already satisfied: tqdm>=4.36.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from fbprophet) (4.50.2)\n",
      "Requirement already satisfied: clikit<0.7,>=0.6 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan>=2.14->fbprophet) (0.6.2)\n",
      "Requirement already satisfied: pysimdjson<4.0,>=3.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan>=2.14->fbprophet) (3.2.0)\n",
      "Requirement already satisfied: aiohttp<4.0,>=3.6 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan>=2.14->fbprophet) (3.7.4.post0)\n",
      "Requirement already satisfied: httpstan<4.6,>=4.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pystan>=2.14->fbprophet) (4.5.0)\n",
      "Requirement already satisfied: pytz>=2017.2 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from pandas>=1.0.4->fbprophet) (2020.1)\n",
      "Requirement already satisfied: certifi>=2020.06.20 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0.0->fbprophet) (2020.6.20)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0.0->fbprophet) (1.3.0)\n",
      "Requirement already satisfied: cycler>=0.10 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0.0->fbprophet) (0.10.0)\n",
      "Requirement already satisfied: pillow>=6.2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0.0->fbprophet) (8.0.1)\n",
      "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from matplotlib>=2.0.0->fbprophet) (2.4.7)\n",
      "Requirement already satisfied: ephem>=3.7.5.3 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from LunarCalendar>=0.0.9->fbprophet) (4.0.0.2)\n",
      "Requirement already satisfied: pymeeus<=1,>=0.3.13 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from convertdate>=2.1.2->fbprophet) (0.5.11)\n",
      "Requirement already satisfied: korean-lunar-calendar in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from holidays>=0.10.2->fbprophet) (0.2.1)\n",
      "Requirement already satisfied: six in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from holidays>=0.10.2->fbprophet) (1.15.0)\n",
      "Requirement already satisfied: hijri-converter in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from holidays>=0.10.2->fbprophet) (2.1.3)\n",
      "Requirement already satisfied: pylev<2.0,>=1.3 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from clikit<0.7,>=0.6->pystan>=2.14->fbprophet) (1.4.0)\n",
      "Requirement already satisfied: pastel<0.3.0,>=0.2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from clikit<0.7,>=0.6->pystan>=2.14->fbprophet) (0.2.1)\n",
      "Requirement already satisfied: crashtest<0.4.0,>=0.3.0; python_version >= \"3.6\" and python_version < \"4.0\" in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from clikit<0.7,>=0.6->pystan>=2.14->fbprophet) (0.3.1)\n",
      "Requirement already satisfied: chardet<5.0,>=2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan>=2.14->fbprophet) (3.0.4)\n",
      "Requirement already satisfied: async-timeout<4.0,>=3.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan>=2.14->fbprophet) (3.0.1)\n",
      "Requirement already satisfied: typing-extensions>=3.6.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan>=2.14->fbprophet) (3.7.4.3)\n",
      "Requirement already satisfied: yarl<2.0,>=1.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan>=2.14->fbprophet) (1.6.3)\n",
      "Requirement already satisfied: multidict<7.0,>=4.5 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan>=2.14->fbprophet) (5.1.0)\n",
      "Requirement already satisfied: attrs>=17.3.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from aiohttp<4.0,>=3.6->pystan>=2.14->fbprophet) (20.3.0)\n",
      "Requirement already satisfied: appdirs<2.0,>=1.4 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan>=2.14->fbprophet) (1.4.4)\n",
      "Requirement already satisfied: webargs<8.0,>=7.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan>=2.14->fbprophet) (7.0.1)\n",
      "Requirement already satisfied: setuptools>=41.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan>=2.14->fbprophet) (50.3.1.post20201107)\n",
      "Requirement already satisfied: marshmallow<4.0,>=3.10 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan>=2.14->fbprophet) (3.12.1)\n",
      "Requirement already satisfied: lz4<4.0,>=3.1 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from httpstan<4.6,>=4.5->pystan>=2.14->fbprophet) (3.1.3)\n",
      "Requirement already satisfied: idna>=2.0 in /Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages (from yarl<2.0,>=1.0->aiohttp<4.0,>=3.6->pystan>=2.14->fbprophet) (2.10)\n",
      "Building wheels for collected packages: fbprophet\n",
      "  Building wheel for fbprophet (setup.py) ... \u001b[?25lerror\n",
      "\u001b[31m  ERROR: Command errored out with exit status 1:\n",
      "   command: /Users/l.sannecompethance.fr/opt/anaconda3/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/setup.py'\"'\"'; __file__='\"'\"'/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' bdist_wheel -d /private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-wheel-8jiijy7w\n",
      "       cwd: /private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/\n",
      "  Complete output (38 lines):\n",
      "  running bdist_wheel\n",
      "  running build\n",
      "  running build_py\n",
      "  creating build\n",
      "  creating build/lib\n",
      "  creating build/lib/fbprophet\n",
      "  creating build/lib/fbprophet/stan_model\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 1, in <module>\n",
      "    File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/setup.py\", line 122, in <module>\n",
      "      setup(\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages/setuptools/__init__.py\", line 153, in setup\n",
      "      return distutils.core.setup(**attrs)\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/core.py\", line 148, in setup\n",
      "      dist.run_commands()\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 966, in run_commands\n",
      "      self.run_command(cmd)\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 985, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages/wheel/bdist_wheel.py\", line 299, in run\n",
      "      self.run_command('build')\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/cmd.py\", line 313, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 985, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/command/build.py\", line 135, in run\n",
      "      self.run_command(cmd_name)\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/cmd.py\", line 313, in run_command\n",
      "      self.distribution.run_command(command)\n",
      "    File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 985, in run_command\n",
      "      cmd_obj.run()\n",
      "    File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/setup.py\", line 48, in run\n",
      "      build_models(target_dir)\n",
      "    File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/setup.py\", line 38, in build_models\n",
      "      StanBackendEnum.get_backend_class(backend).build_model(target_dir, MODEL_DIR)\n",
      "    File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/fbprophet/models.py\", line 209, in build_model\n",
      "      import pystan\n",
      "  ModuleNotFoundError: No module named 'pystan'\n",
      "  ----------------------------------------\u001b[0m\n",
      "\u001b[31m  ERROR: Failed building wheel for fbprophet\u001b[0m\n",
      "\u001b[?25h  Running setup.py clean for fbprophet\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Failed to build fbprophet\n",
      "Installing collected packages: fbprophet\n",
      "    Running setup.py install for fbprophet ... \u001b[?25lerror\n",
      "\u001b[31m    ERROR: Command errored out with exit status 1:\n",
      "     command: /Users/l.sannecompethance.fr/opt/anaconda3/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/setup.py'\"'\"'; __file__='\"'\"'/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record /private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-record-5ocipwgg/install-record.txt --single-version-externally-managed --compile --install-headers /Users/l.sannecompethance.fr/opt/anaconda3/include/python3.8/fbprophet\n",
      "         cwd: /private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/\n",
      "    Complete output (40 lines):\n",
      "    running install\n",
      "    running build\n",
      "    running build_py\n",
      "    creating build\n",
      "    creating build/lib\n",
      "    creating build/lib/fbprophet\n",
      "    creating build/lib/fbprophet/stan_model\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/setup.py\", line 122, in <module>\n",
      "        setup(\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages/setuptools/__init__.py\", line 153, in setup\n",
      "        return distutils.core.setup(**attrs)\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/core.py\", line 148, in setup\n",
      "        dist.run_commands()\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 966, in run_commands\n",
      "        self.run_command(cmd)\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 985, in run_command\n",
      "        cmd_obj.run()\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/site-packages/setuptools/command/install.py\", line 61, in run\n",
      "        return orig.install.run(self)\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/command/install.py\", line 545, in run\n",
      "        self.run_command('build')\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/cmd.py\", line 313, in run_command\n",
      "        self.distribution.run_command(command)\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 985, in run_command\n",
      "        cmd_obj.run()\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/command/build.py\", line 135, in run\n",
      "        self.run_command(cmd_name)\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/cmd.py\", line 313, in run_command\n",
      "        self.distribution.run_command(command)\n",
      "      File \"/Users/l.sannecompethance.fr/opt/anaconda3/lib/python3.8/distutils/dist.py\", line 985, in run_command\n",
      "        cmd_obj.run()\n",
      "      File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/setup.py\", line 48, in run\n",
      "        build_models(target_dir)\n",
      "      File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/setup.py\", line 38, in build_models\n",
      "        StanBackendEnum.get_backend_class(backend).build_model(target_dir, MODEL_DIR)\n",
      "      File \"/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/fbprophet/models.py\", line 209, in build_model\n",
      "        import pystan\n",
      "    ModuleNotFoundError: No module named 'pystan'\n",
      "    ----------------------------------------\u001b[0m\n",
      "\u001b[31mERROR: Command errored out with exit status 1: /Users/l.sannecompethance.fr/opt/anaconda3/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/setup.py'\"'\"'; __file__='\"'\"'/private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-install-jwbwosot/fbprophet/setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record /private/var/folders/1t/6j3bfwfx0vz_n8yvd_pth3zw0000gn/T/pip-record-5ocipwgg/install-record.txt --single-version-externally-managed --compile --install-headers /Users/l.sannecompethance.fr/opt/anaconda3/include/python3.8/fbprophet Check the logs for full command output.\u001b[0m\n",
      "\u001b[?25h"
     ]
    }
   ],
   "source": [
    "!pip install fbprophet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'fbprophet'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-13-25ad8bc6c860>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mplotly\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfigure_factory\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mff\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mplotly\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mgraph_objs\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mgo\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 9\u001b[0;31m \u001b[0;32mfrom\u001b[0m \u001b[0mfbprophet\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mProphet\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     10\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mfbprophet\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mplot\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mplot_plotly\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'fbprophet'"
     ]
    }
   ],
   "source": [
    "#Importation des librairies nécessaires\n",
    "\n",
    "import streamlit as st\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import chart_studio.plotly as plotly\n",
    "import plotly.figure_factory as ff\n",
    "from plotly import graph_objs as go\n",
    "from fbprophet import Prophet\n",
    "from fbprophet.plot import plot_plotly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "st.title('App Prix Production de tomates')\n",
    "\n",
    "dataset = ('TM','TM2')\n",
    "\n",
    "option = st.selectbox('Choisir le dataset pour les prédictions',dataset)\n",
    "DATA_URL =('./DATA/'+option+'.csv')\n",
    "\n",
    "Année = st.slider('Année pour les prédicitions:',1,8)\n",
    "Jour = st.slider('Jour pour les prédictions:',1,30)\n",
    "\n",
    "@st.cache\n",
    "def load_data():\n",
    "    data = pd.read_csv(DATA_URL)\n",
    "    return data\n",
    "\n",
    "\n",
    "\n",
    "data_load_state = st.text('Téléchargement des données...')\n",
    "data = load_data()\n",
    "data_load_state.text('Téléchargement des données... terminé!')\n",
    "\n",
    "def plot_fig():\n",
    "    fig = go.Figure()\n",
    "    fig.add_trace(go.Scatter(x=data.Date, y=data['prix moyen au kg'], name=\"prix moyen au kg\",line_color='darkviolet'))\n",
    "    fig.add_trace(go.Scatter(x=data.Date, y=data['Production quantité \\ntonne(s)'], name=\"Production quantité \\ntonne(s)\",line_color='gold'))\n",
    "    fig.layout.update(title_text='Time Series data with Rangeslider',xaxis_rangeslider_visible=True)\n",
    "    st.plotly_chart(fig)\n",
    "    return fig\n",
    "\n",
    "if st.checkbox('Show raw data'):\n",
    "    st.subheader('Raw data')\n",
    "    \n",
    "plot_fig()\n",
    "\n",
    "\n",
    "data_pred1 = data[['Date','prix moyen au kg']]\n",
    "data_pred1['ds'] = pd.to_datetime(data_pred1['Date'])\n",
    "data_pred1['y'] = data_pred1['prix moyen au kg']\n",
    "data_pred1 = data_pred1.drop([\"Date\",\"prix moyen au kg\"], axis=1)\n",
    "\n",
    "data_pred2 = data[['Date','Production quantité \\ntonne(s)']]\n",
    "data_pred2['ds'] = pd.to_datetime(data_pred2['Date'])\n",
    "data_pred2['y'] = data_pred2['Production quantité \\ntonne(s)']\n",
    "data_pred2 = data_pred2.drop([\"Date\",\"Production quantité \\ntonne(s)\"], axis=1)\n",
    "\n",
    "m = Prophet(daily_seasonality=True)\n",
    "m.fit(data_pred1)\n",
    "future = m.make_future_dataframe(periods=7, freq='D')\n",
    "forecast = m.predict(future)\n",
    "\n",
    "\n",
    "m2 = Prophet(daily_seasonality=True)\n",
    "m2.fit(data_pred2)\n",
    "future2 = m2.make_future_dataframe(periods=7, freq='D')\n",
    "forecast2 = m.predict(future2)\n",
    "\n",
    "\n",
    "\n",
    "fig1 = plot_plotly(m, forecast)\n",
    "if st.checkbox('Show forecast data'):\n",
    "    st.subheader('forecast data')\n",
    "    st.write(forecast)\n",
    "st.write('Forecasting closing of stock value for'+option+' pour une période de: '+str(Jour)+'Jour')\n",
    "st.plotly_chart(fig1)\n",
    "\n",
    "fig2 = plot_plotly(m2, forecast2)\n",
    "if st.checkbox('Show forecast data'):\n",
    "    st.subheader('forecast data')\n",
    "    st.write(forecast2)\n",
    "st.write('Forecasting closing of stock value for'+option+' pour une période de: '+str(Jour)+'Jour')\n",
    "st.plotly_chart(fig2)\n",
    "\n",
    "\n",
    "st.write(\"Component wise forecast\")\n",
    "fig3 = m.plot_components(forecast)\n",
    "st.write(fig3)\n",
    "\n",
    "st.write(\"Component wise forecast\")\n",
    "fig4 = m2.plot_components(forecast2)\n",
    "st.write(fig4)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "numpy==1.17.0\n",
    "streamlit==0.49.0\n",
    "plotly==3.10.0\n",
    "pandas==0.25.0\n",
    "chart_studio==1.0.0\n",
    "fbprophet==0.5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
